app/code/Magento/Catalog/Model/ResourceModel/Category/Collection.php

```diff
@@ -404,6 +462,23 @@ public function addRootLevelFilter()
         return $this;
     }
 
+    /**
+     * Add navigation max depth filter
+     *
+     * @return $this
+     */
+    public function addNavigationMaxDepthFilter()
+    {
+        $navigationMaxDepth = (int)$this->scopeConfig->getValue(
+            'catalog/navigation/max_depth',
+            ScopeInterface::SCOPE_STORE
+        );
+        if ($navigationMaxDepth > 0) {
+            $this->addLevelFilter($navigationMaxDepth);
```

Comment:
In my test, this looks like it doesn't work as intended. If I set a maximum depth of 1, no category is shown. If I set the value to 2, the top level category is displayed. If we add 1 to `$navigationMaxDepth` like `$this->addLevelFilter($navigationMaxDepth + 1)`, it works like I would expect it to. I guess that's because the root category level also has to be considered.

Reply:
No, that seems not like a good solution to me. I think we should adhere to the levels on http://docs.magento.com/m2/ee/user_guide/catalog/navigation-top.html

So in the admin configuration you have to use 2 for the first level.

Reply:
You are absolutely right. Thank you for pointing this out. The PR works as intended.